tron/serialize/runstate/dynamodb_state_store.py

```diff
@@ -129,28 +138,45 @@ def _get_remaining_partitions(self, items: list):
         for item in items:
             remaining_items = [
                 {"key": {"S": str(item["key"]["S"])}, "index": {"N": str(i)}}
-                for i in range(1, int(item["num_partitions"]["N"]))
+                for i in range(1, int(max(item["num_partitions"]["N"], item["num_json_val_partitions"]["N"])) + 1)
             ]
             keys_for_remaining_items.extend(remaining_items)
         return self._get_items(keys_for_remaining_items)
 
-    def _merge_items(self, first_items, remaining_items) -> dict:
+    # read_json is a flag set in srv-configs to read json data or not, otherwise will read pickled data
+    def _merge_items(self, first_items, remaining_items, read_json=False) -> dict:
         items = defaultdict(list)
         raw_items: DefaultDict[str, bytearray] = defaultdict(bytearray)
+        json_items: DefaultDict[str, str] = defaultdict(str)
+        # item = job run, each job run can have multiple rows and each row is called a partition
         # Merge all items based their keys and deserialize their values
         if remaining_items:
             first_items.extend(remaining_items)
         for item in first_items:
-            key = item["key"]["S"]
-            items[key].append(item)
+            key = item["key"]["S"]  # key column of an item = "job_State MASTER.cits_test_load_foo1"
+            items[key].append(item)  # for jobs state one item is basically info for each job: run nums and enabled
         for key, item in items.items():
             item.sort(key=lambda x: int(x["index"]["N"]))
             for val in item:
                 raw_items[key] += bytes(val["val"]["B"])
-        deserialized_items = {k: pickle.loads(v) for k, v in raw_items.items()}
+            for json_val in item:
+                json_items[key] = json_val["json_val"]["S"]
```

Comment:
think it's worth gating this behind `read_json` as well? in case we need to turn json reading and writing off (and thus there's no `json_val`)